utils/missingflowstracker.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"sync"
+)
+
+// MissingFlowsTracker is used to track missing packets/flows
+type MissingFlowsTracker struct {
+	counters   map[string]int64 // counter key is based on source addr and sourceId/obsDomain/engineType/engineId
+	countersMu *sync.RWMutex
+
+	maxNegativeSequenceDifference int
+}
+
+func NewMissingFlowsTracker(maxNegativeSequenceDifference int) *MissingFlowsTracker {
+	return &MissingFlowsTracker{
+		counters:                      make(map[string]int64),
+		countersMu:                    &sync.RWMutex{},
+		maxNegativeSequenceDifference: maxNegativeSequenceDifference,
+	}
+}
+
+func (s *MissingFlowsTracker) countMissing(key string, seqnum uint32, flows uint16) int64 {
+	s.countersMu.Lock()
+	defer s.countersMu.Unlock()
+
+	if _, ok := s.counters[key]; !ok {
+		s.counters[key] = int64(seqnum)
+	} else {
+		s.counters[key] += int64(flows)
+	}
+	missingElements := int64(seqnum) - s.counters[key]
+
+	// We assume there is a sequence number reset when the number of missing flows/packets is negative and high.
+	// When this happens, we reset the counter to the current sequence number.
+	if missingElements <= -int64(s.maxNegativeSequenceDifference) {
```

Comment:
@ken-schneider 
`maxNegativeSequenceDifference` can be indeed configured, but it has only 2 possible values:
- value of 100 for netflow9/ipfix that tracks number of packets.
- value of 1000 for netflow5 that tracks number of flows.

Taking your example, for netflow9, an issue like the one you described might happen if chunks of 100+ packets are lost at time. About the likelihood, I think it's possible to see drop of 100+ packet if there are some network issues for example, but the likelihood should be very low to see as consistently as your example with P1/P2/P3/P4.

Also in such environment with high volume of drops, what can happen is that we are not counting some missing flows, due to this reset mechanism.
But I think in practice is shouldn't be that a big issue since we mainly want to know if we are dropping significant number of flows, but we are less interested in the exact/precise number of flows being drops.

About those 2 constants, I intuitively choose 100 packets and 1000 flows, but there are possibly better number and we might also want to make them configurable in the future if needed:
```
var MaxNegativePacketsSequenceDifference = 100
var MaxNegativeFlowsSequenceDifference = 1000
```